apps/survey/test/survey.js

```diff
@@ -6,25 +6,53 @@ const getContract = name => artifacts.require(name)
 const pct16 = x => new web3.BigNumber(x).times(new web3.BigNumber(10).toPower(16))
 const createdSurveyId = receipt => receipt.logs.filter(x => x.event == 'StartSurvey')[0].args.surveyId
 
+const ANY_ADDR = '0xffffffffffffffffffffffffffffffffffffffff'
+
 contract('Survey app', accounts => {
-  let survey, ABSTAIN_VOTE
+  let daoFact, surveyBase, survey
+
+  let ABSTAIN_VOTE, APP_MANAGER_ROLE, CREATE_SURVEYS_ROLE, MODIFY_PARTICIPATION_ROLE
+
   const surveyTime = 1000
+  const root = accounts[0]
   const holder19 = accounts[0]
   const holder31 = accounts[1]
   const holder50 = accounts[2]
   const nonHolder = accounts[4]
   const NULL_ADDRESS = '0x00'
+  const minimumAcceptanceParticipationPct = pct16(20)
+
+  before(async () => {
+    const kernelBase = await getContract('Kernel').new(true) // petrify immediately
+    const aclBase = await getContract('ACL').new()
+    const regFact = await getContract('EVMScriptRegistryFactory').new()
```

Comment:
One small nitpick: I don't think we need `EVMScripts` with Survey, so not using one in the `DAOFactory` will speed up the tests.